src/mysql.c

```diff
@@ -973,6 +977,31 @@ static int mysql_read (user_data_t *ud)
 				counter_submit ("mysql_innodb_rows", "read", val, db);
 			else if (strcmp (key, "Innodb_rows_updated") == 0)
 				counter_submit ("mysql_innodb_rows", "updated", val, db);
+
+ 			/* checkpoints */
+ 			else if (strcmp (key, "Innodb_checkpoint_age") == 0)
+ 				gauge_submit ("mysql_innodb_checkpoint", "age", val, db);
+ 			else if (strcmp (key, "Innodb_checkpoint_max_age") == 0)
+ 				gauge_submit ("mysql_innodb_checkpoint", "max_age", val, db);
+ 
+ 			/* max id */
+ 			else if (strcmp (key, "Innodb_max_trx_id") == 0)
+ 				gauge_submit ("mysql_innodb_max", "trx_id", val, db);
```

Comment:
The type *mysql_innodb_max* odes not exist. Assuming that the transaction ID is an integer that is increased by one for each new transaction, I would propose to create a new type like this:

```
total_transactions value:DERIVE:0:U
```